lib/Common/Core/RecyclerVisitedObject.h

```diff
@@ -0,0 +1,31 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+interface IRecyclerHeapMarkingContext;
+typedef void* RecyclerHeapHandle;
+
+interface IRecyclerVisitedObject
```

Comment:
>interface [](start = 0, length = 9)

Curious if "interface" keyword is supported by other compilers? Also what's the reason you changed Finalize/Dispose to __stdcall across the board? Is it required by "interface"?

Wonder if it works by just using abstract c++ class. That'll probably look more standard compliant and less noisy in this PR.